include/Dialects/LinalgTransform/LinalgTransformOps.td

```diff
@@ -33,6 +34,34 @@ class Linalg_Transform_Operation<string name, list<OpTrait> props = []>
 
 //===----------------------------------------------------------------------===//
 
+def ScopeOp : Linalg_Transform_Operation<"scope",
```

Comment:
We were discussing doing exactly this but on FuncOps, with the extra benefit of being able to "name" things.
So I started slicing out functionality (e.g. https://reviews.llvm.org/D116591).

Looking forward, do you see value in:
1. keeping this op without a Symbol interface ?
2. having a separate op than just FuncOp ?

Reply:
This operation is meant to be short-lived. I.e.

```
ScopeOp scope = wrapInScope(myOp);
/// do a bunch of transformations in the scope
unwrapScope(scope);
```

In which case it would not need a symbol to look up later. If it needs to exist longer than a few transformations, then you could pass the handle around instead of passing the name around and looking it up.